account_payment_credit_card/views/account_view.xml

```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+
+    <record id="view_account_journal_creditcard_form" model="ir.ui.view">
+        <field name="name">account.journal.form</field>
+        <field name="model">account.journal</field>
+        <field name="type">form</field>
+        <field name="inherit_id" ref="account.view_account_journal_form" />
+        <field name="arch" type="xml">
+            <field name="default_account_id" position="after">
+                <field
+                    name="credit_card"
+                    attrs="{'readonly':[('type', '!=', 'bank')]}"
```

Comment:
Maybe it is not necessary to show this fields if it not journal type bank

![image](https://github.com/user-attachments/assets/1fcfb108-1c42-415a-9fa4-b71a210ba479)